python/lsst/daf/butler/core/registry.py

```diff
@@ -182,8 +203,39 @@ def fromConfig(registryConfig, schemaConfig=None, dimensionConfig=None, create=F
             else:
                 raise ValueError("Incompatible Registry configuration: {}".format(registryConfig))
 
-        cls = doImport(registryConfig["cls"])
-        return cls(registryConfig, schemaConfig, dimensionConfig, create=create, butlerRoot=butlerRoot)
+        cls = Registry.getRegistryClass(registryConfig)
+
+        return cls(registryConfig, schemaConfig, dimensionConfig, create=create,
+                   butlerRoot=butlerRoot)
+
+    @classmethod
+    def getRegistryClass(cls, registryConfig):
+        """Returns registry class targeted by configuration values.
+
+        The appropriate class is determined from the `cls` key, if it exists.
+        Otherwise the `db` key is parsed and the correct class is determined
+        from a list of aliases found under `clsMap` key of the registry config.
+
+        Parameters
+        ----------
+        registryConfig : `ButlerConfig`, `RegistryConfig`, `Config` or `str`
+            Registry configuration
+
+        Returns
+        -------
+        registry : `Registry` (subclass)
```

Comment:
I think it returns a `type` of `Registry` not an instance of `Registry`.

Reply:
A tongue twister but better?

Reply:
Maybe say "class of type `Registry`" ?

Reply:
Sorted, 

> class of type `Registry`

it is.